File: skellig-test-step-processing/src/main/java/org/skellig/teststep/processing/model/TestStepExecutionType.java
package org.skellig.teststep.processing.model;

public enum TestStepExecutionType {
    SYNC("sync"),
    ASYNC("async");

    private String name;

    TestStepExecutionType(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static TestStepExecutionType fromName(String name) {
        for (TestStepExecutionType value : values()) {
            if (value.name.equalsIgnoreCase(name)) {
                return value;
            }
        }
        return TestStepExecutionType.SYNC;
    }
}

File: skellig-test-step-processing/src/main/kotlin/org/skellig/teststep/processing/model/TestStepExecutionType.kt
package org.skellig.teststep.processing.model

/**
 * Represents the execution type of test step.
 * This type is used by [TestStepProcessor][org.skellig.teststep.processing.processor.TestStepProcessor] to determine
 * the way of running a test step: synchronously or asynchronously.
 */
enum class TestStepExecutionType {
    SYNC,
    ASYNC;

    companion object {
        fun fromName(name: String?): TestStepExecutionType {
            for (value in entries) {
                if (value.name.equals(name, ignoreCase = true)) {
                    return value
                }
            }
            return SYNC
        }
    }
}